src/app/components/Ad/index.tsx

```diff
@@ -1,13 +1,30 @@
 import React, { useContext } from 'react';
 import useToggle from '#hooks/useToggle';
 import { RequestContext } from '#contexts/RequestContext';
+import crypto from 'crypto';
+import { ServiceContext } from '#app/contexts/ServiceContext';
 import AmpAd from './Amp';
 import CanonicalAd from './Canonical';
 import { AdProps } from './types';
 import AdBootstrapJs from './Canonical/AdBootstrapJs';
 
+const servicesForNonce = ['mundo', 'arabic'];
+const countriesForNonce = ['eg', 'mx'];
+
 const AdContainer = ({ slotType, className, adcampaign }: AdProps) => {
-  const { isAmp, isLite, showAdsBasedOnLocation } = useContext(RequestContext);
+  const { isAmp, isLite, showAdsBasedOnLocation, country } =
+    useContext(RequestContext);
+
+  const { service } = useContext(ServiceContext);
+  const isNonceAllowed =
+    servicesForNonce.includes(service) &&
+    country &&
+    countriesForNonce.includes(country);
```

Comment:
We can remove the checks for `country` since `countriesForNonce.includes(country)` would return `false` if country was `null` or `undefined`
```suggestion
  const isNonceAllowed =
    servicesForNonce.includes(service) &&
    countriesForNonce.includes(country);
```

Reply:
<img width="1170" alt="Screenshot 2025-06-17 at 17 05 19" src="https://github.com/user-attachments/assets/cf3eda0f-02c9-4313-9be2-7756f566e5df" />

Reply:
@karinathomasbbc

Reply:
Ah shame. OK thanks for letting me know!